tests/components/eheimdigital/test_light.py

```diff
@@ -0,0 +1,232 @@
+
+    await hass.services.async_call(
+        LIGHT_DOMAIN,
+        SERVICE_TURN_ON,
+        {
+            ATTR_ENTITY_ID: "light.mock_classicledcontrol_e_channel_0",
+            ATTR_BRIGHTNESS: dim_input,
+        },
+        blocking=True,
+    )
+
+    classic_led_ctrl_mock.set_light_mode.assert_awaited_once_with(LightMode.MAN_MODE)
+    classic_led_ctrl_mock.turn_on.assert_awaited_once_with(expected_dim_value, 0)
+
+
+async def test_turn_on_effect(
+    hass: HomeAssistant,
+    eheimdigital_hub_mock: MagicMock,
+    mock_config_entry: MockConfigEntry,
+    classic_led_ctrl_mock: MagicMock,
+) -> None:
+    """Test turning on the light with an effect value."""
+    mock_config_entry.add_to_hass(hass)
+
+    eheimdigital_hub_mock.return_value.devices = {
+        "00:00:00:00:00:01": classic_led_ctrl_mock
+    }
+    eheimdigital_hub_mock.return_value.master = classic_led_ctrl_mock
+
+    classic_led_ctrl_mock.light_mode = LightMode.MAN_MODE
+
+    with patch("homeassistant.components.eheimdigital.PLATFORMS", [Platform.LIGHT]):
+        await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await eheimdigital_hub_mock.call_args.kwargs["device_found_callback"](
+        "00:00:00:00:00:01", EheimDeviceType.VERSION_EHEIM_CLASSIC_LED_CTRL_PLUS_E
+    )
+    await hass.async_block_till_done()
+
+    await hass.services.async_call(
+        LIGHT_DOMAIN,
+        SERVICE_TURN_ON,
+        {
+            ATTR_ENTITY_ID: "light.mock_classicledcontrol_e_channel_0",
+            ATTR_EFFECT: EFFECT_DAYCL_MODE,
+        },
+        blocking=True,
+    )
+
+    classic_led_ctrl_mock.set_light_mode.assert_awaited_once_with(LightMode.DAYCL_MODE)
+
+
+async def test_state_update(
+    hass: HomeAssistant,
+    eheimdigital_hub_mock: MagicMock,
+    mock_config_entry: MockConfigEntry,
+    entity_registry: er.EntityRegistry,
+    snapshot: SnapshotAssertion,
+    classic_led_ctrl_mock: MagicMock,
+) -> None:
+    """Test the light state update."""
+    mock_config_entry.add_to_hass(hass)
+
+    eheimdigital_hub_mock.return_value.devices = {
+        "00:00:00:00:00:01": classic_led_ctrl_mock
+    }
+    eheimdigital_hub_mock.return_value.main = classic_led_ctrl_mock
+
+    with patch("homeassistant.components.eheimdigital.PLATFORMS", [Platform.LIGHT]):
+        await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await eheimdigital_hub_mock.call_args.kwargs["device_found_callback"](
+        "00:00:00:00:00:01", EheimDeviceType.VERSION_EHEIM_CLASSIC_LED_CTRL_PLUS_E
+    )
+    await hass.async_block_till_done()
+
+    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
+
+    classic_led_ctrl_mock.light_level = (20, 30)
+
+    await mock_config_entry.runtime_data._async_receive_callback()
+
+    assert (state := hass.states.get("light.mock_classicledcontrol_e_channel_0"))
+    assert state == snapshot(name="light.mock_classicledcontrol_e_channel_0-state-2")
```

Comment:
Ideally we don't use a snapshot test here since you are looking for something specific